File: CSharp/OpenTraceability/Interfaces/IILMDEvent.cs
using OpenTraceability.Models.Events;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenTraceability.Interfaces
{
    public interface IILMDEvent<T> : IEvent where T : EventILMD
    {
        T ILMD { get; set; }
    }
}

File: Java/opentraceability/src/main/java/opentraceability/interfaces/IILMDEvent.java
package opentraceability.interfaces;

import opentraceability.models.events.EPCISVersion;
import opentraceability.models.events.EventILMD;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityObjectAttribute;

public abstract class IILMDEvent<T extends EventILMD> extends IEvent {
    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute(ns = "", name = "ilmd", sequenceOrder = 50, version = EPCISVersion.V2)
    @OpenTraceabilityAttribute(ns = "", name = "extension/ilmd", sequenceOrder = 50, version = EPCISVersion.V1)
    public T ilmd = null;
}